controllers/front/CommentGrade.php

```diff
@@ -29,17 +29,27 @@ class ProductCommentsCommentGradeModuleFrontController extends ModuleFrontContro
 {
     public function display()
     {
-        $idProduct = Tools::getValue('id_product');
+        $idProducts = Tools::getValue('id_products');
         /** @var ProductCommentRepository $productCommentRepository */
+
+
         $productCommentRepository = $this->context->controller->getContainer()->get('product_comment_repository');
 
-        $productCommentsNb = $productCommentRepository->getCommentsNumber($idProduct, Configuration::get('PRODUCT_COMMENTS_MODERATE'));
-        $averageGrade = $productCommentRepository->getAverageGrade($idProduct, Configuration::get('PRODUCT_COMMENTS_MODERATE'));
+        $productsCommentsNb = $productCommentRepository->getCommentsNumberForProducts($idProducts, Configuration::get('PRODUCT_COMMENTS_MODERATE'));
+        $averageGrade = $productCommentRepository->getAverageGrades($idProducts, Configuration::get('PRODUCT_COMMENTS_MODERATE'));
+
+        $resultFormated = [];
+
+        foreach ($idProducts as $i => $id) {
```

Comment:
$idProducts is not always an array

Reply:
@PierreRambaud 

How is it not always an array?
In productListingComments.js on line 114 we are sending array of ids.

Reply:
But you need to check if it's an array.
The `Tools::getValue` can return something different

Reply:
Check added